### ISSUE TITLE
command calls does not seem to have basic errorhandling in place

### ISSUE THREAD

**Issue by jborup** [openviess/PyViCare#152]
Hi,

I tried doing a `print(t.getOneTimeCharge())` and I was a little surprised when it just returned:
{"viErrorId":"req-XXXXX","statusCode":502,"errorType":"DEVICE_COMMUNICATION_ERROR","message":"DEVICE_COMMUNICATION_ERROR","extendedPayload":{"reason":"INTERNAL_SERVER_ERROR"}}

But no exceptions was thrown.

Looks like a @handleAPIErrors or something similar at set calls like activateOneTimeCharg() could be a great improvement to know it fails, rather than having to parse the response and detect it is in error.

Any thoughts?

**Comment by woehrl01** [comment]
Good idea!

**Comment by woehrl01** [comment]
You mentioned getOneTimeCharge and activateOneTimeCharge. Which was the exact ohne you have used? As the get call should not resolute in a side effect if activating something and thus should not return your described error.

**Comment by jborup** [comment]
I created a PR #153 so in response to your comment maybe you see the answer in the code. What triggered me was the activeOneTimeCharge() function not handling errors. So I "thought" things was okay until I started doing the postman equivalent and saw it was in error. Now any post calls leading to statuscode 4xx, 5xx will lead to this exception. Which I believe should be an okay exception in general to a POST call not returning 2xx or 3xx.

If any comments feel free, if not suggest to just merge the PR :-)
